src/box/space_def.c

```diff
@@ -359,10 +361,60 @@ space_opts_parse_type(const char **data, void *vopts, struct region *region)
 		if (space_type != SPACE_TYPE_NORMAL)
 			goto conflict;
 		break;
+	case SPACE_TYPE_META_TEMPORARY:
+		/*
+		 * This means type was specified twice, which can happen if the
+		 * tuple was inserted directly, e.g. via ffi api. This is an
+		 * unlikely case, but we shouldn't panic either way. Other
+		 * options decoding code handles this case by only respecting
+		 * the last occurrence of the option. So does this code.
+		 */
+		opts->type = space_type;
+		break;
 	}
 	return 0;
 
 conflict:
 	diag_set(IllegalParams, "'type' conflicts with 'temporary'");
 	return -1;
 }
+
+bool
+space_def_tuple_is_meta_temporary(const char *data, uint32_t *space_id)
+{
+	uint32_t field_count = mp_decode_array(&data);
+	if (field_count < BOX_SPACE_FIELD_OPTS + 1)
+		return false;
+
+	static_assert(BOX_SPACE_FIELD_ID == 0,
+		      "the following code relies on this assumption");
+	if (mp_typeof(*data) != MP_UINT)
+		return false;
+	uint32_t id = mp_decode_uint(&data);
+	if (space_id)
```

Comment:
It is always not `NULL`. Please, drop the `if`.

Reply:
done